integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -701,13 +706,14 @@ private int releaseInternal(String path, FuseFileInfo fi) {
     OpenFileEntry oe;
     final long fd = fi.fh.get();
     oe = mOpenFiles.getFirstByField(ID_INDEX, fd);
-    mOpenFiles.remove(oe);
-    if (oe == null) {
+    if (oe == null || !mOpenFiles.remove(oe)) {
```

Comment:
Does the `mOpenFiles.remove(oe)` need to be included in the synchronized(oe)?
I am not 100% sure if the following scenario will happen:
1) read -> enter the synchronized(oe) block, check the mOpenFiles contains the oe and start reading
2) during the reading, release() is called and oe is removed from mOpenFiles.

The above case seems unlikely to happen, but to be safe, can we move the logic into the synchronized block? 

what's your thoughts?

Reply:
I think the  above scenario will happen, but it is safe, because even `oe` is removed, read thread can also read the data from alluxio if the read thread enter the synchronized block. For the release thread, it can close the stream only after enter the synchronized block.

Reply:
Make sense, then i am fine with keeping the original logics